internal/services/dns/dns_zone_resource.go

```diff
@@ -180,13 +180,27 @@ func resourceDnsZoneCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) er
 		return fmt.Errorf("creating/updating %s: %+v", id, err)
 	}
 
+	d.SetId(id.ID())
+
 	if v, ok := d.GetOk("soa_record"); ok {
 		soaRecord := v.([]interface{})[0].(map[string]interface{})
+
+		inputSOARecord := expandArmDNSZoneSOARecord(soaRecord)
+
+		if *inputSOARecord.Host == "" {
+			resourceDnsZoneRead(d, meta)
```

Comment:
Instead of calling the `read` function here, can we just call the `GET` on the SOA record to avoid the side effect from calling the `read`?

Reply:
I will investigate that

Reply:
I think this is fix, please check if I correctly understood your ask.